samples/key-value-customer-registry/src/main/java/customer/api/CustomerEndpoint.java

```diff
@@ -108,31 +110,67 @@ public HttpResponse getCustomersCsvByName(String name) {
-    // Server sent events
-    // https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events#event_stream_format
-    Source<ByteString, NotUsed> sseCustomerSummaryStream =
-        customerSummarySource.map(customerSummary ->
-            eventPrefix.concat(JsonSupport.encodeToAkkaByteString(customerSummary)).concat(eventEnd)
+    return HttpResponses.serverSentEvents(customerSummarySource);
+  }
+  
+  @Get("/stream-customer-changes/{customerId}")
+  public HttpResponse streamCustomerChanges(String customerId) {
+    Source<Customer, Cancellable> stateEvery5Seconds =
+        // stream of ticks, one immediately, then one every five seconds
+        Source.tick(Duration.ZERO, Duration.ofSeconds(5), "tick")
+          // for each tick, request the entity state
+          .mapAsync(1, __ ->
+            componentClient.forKeyValueEntity(customerId)
+                .method(CustomerEntity::getCustomer)
+                .invokeAsync().handle((Customer customer, Throwable error) -> {
+                  if (error == null) {
+                    return Optional.of(customer);
+                  } else if (error instanceof IllegalArgumentException) {
+                    // calling getCustomer throws IllegalArgument if the customer does not exist
+                    // we want the stream to continue polling in that case, so turn it into an empty optional
+                    return Optional.<Customer>empty();
+                  } else {
+                    throw new RuntimeException("Unexpected error polling customer state", error);
+                  }
+                })
+          )
+          // then filter out the empty optionals and return the actual customer states for nonempty
+          // so that the stream contains only Customer elements
+          .filter(Optional::isPresent).map(Optional::get);
+
+    // deduplicate, so that we don't emit if the state did not change from last time
+    Source<Customer, Cancellable> streamOfChanges = stateEvery5Seconds.statefulMapConcat(() -> {
+      // Java does not allow mutating state in surrounding scope, so we need an array for that
+      Customer[] previousCustomer = new Customer[1];
+
+      return (Customer customer) -> {
+        if (previousCustomer[0] == null || !previousCustomer[0].equals(customer)) {
+          // there was no previous customer, or the new state is different from what was previously
+          // seen, emit the customer
+          previousCustomer[0] = customer;
+          return Collections.singletonList(customer);
+        } else {
+          // the state was identical to previous, emit nothing
+          return Collections.emptyList();
+        }
+      };
+    });
+
+    // now turn each changed internal state representation into public API representation,
+    // just like get endpoint above
+    Source<CustomerRequest, Cancellable> streamOfChangesAsApiType = streamOfChanges.map(c ->
```

Comment:
confusing that the output is `CustomerRequest`, which is named like an input type

Reply:
Yeah I thought so as well, I'll change that.